### PR TITLE
post: gmail-wrapper open-source

### PR DESCRIPTION
**Pull request by luizguilhermefr**

### CODE REVIEW

**Comment by barbosa** [comment]
@luizguilhermefr would you mind rebasing this branch with master? Just so that we have a clear preview on Netlify.

**Comment by luizguilhermefr** [comment]
> @luizguilhermefr would you mind rebasing this branch with master? Just so that we have a clear preview on Netlify.

@barbosa done

File: content/posts/2020-07-01-introducing-gmail-wrapper/index.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+author: Luiz Rosa
+twitter: luizguilhermefr
+layout: post
+lang: en
+path: /blog/introducing-gmail-wrapper
+date: '2020-07-22'
+comments: true
+---
+
+If you ever had to read or write emails using the Gmail Official [Python SDK](https://googleapis.github.io/google-api-python-client/docs/dyn/gmail_v1.html) you know it is a pain for some reasons:
```
Comment (alexantoniomoran) [inline content/posts/2020-07-01-introducing-gmail-wrapper/index.md:12]:
```suggestion
If you've ever had to read or write emails using the Gmail Official [Python SDK](https://googleapis.github.io/google-api-python-client/docs/dyn/gmail_v1.html), then you know it is a pain for multiple reasons:
```

File: content/posts/2020-07-01-introducing-gmail-wrapper/index.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+- It is verbose, using dozens of chained objects like `foo().bar().baz()`
+- It is not intuitive (How to authenticate? How to read my messages? How to send a message? I don’t know...)
```
Comment (alexantoniomoran) [inline content/posts/2020-07-01-introducing-gmail-wrapper/index.md:15]:
```suggestion
- It is not intuitive (How to I authenticate? How to I read my messages? How do I send a message? I don’t know...)
```

File: content/posts/2020-07-01-introducing-gmail-wrapper/index.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+- Although using a lot of classes, the responses are always plain dictionaries (What are the fields? I’ll have to look the docs again)
```
Comment (alexantoniomoran) [inline content/posts/2020-07-01-introducing-gmail-wrapper/index.md:16]:
```suggestion
- It uses plain dictionaries as the default response, regardless of the number of classes used (What are the fields? I’ll have to look the docs again)

```

File: content/posts/2020-07-01-introducing-gmail-wrapper/index.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+We use Gmail for several automated tasks at Loadsmart, and to overcome these difficulties we developed [gmail-wrapper](https://github.com/loadsmart/gmail-wrapper), a Python library that encapsulates the hard parts of the official SDK and provides an easier API.
+It was a closed project used only internally, and now we decided to bring it open-source, the first Loadsmart open Python package.
```
Comment (alexantoniomoran) [inline content/posts/2020-07-01-introducing-gmail-wrapper/index.md:19]:
```suggestion
This gmail-wrapper was a closed project used only internally, but now we decided to make it the first Loadsmart open-sourced Python package.
```

File: content/posts/2020-07-01-introducing-gmail-wrapper/index.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+```python
+messages = client.get_messages(query="receipts", limit=10)
+for message in messages:
+    print(message.subject) # Gmail Wrapper will automatically lazy-load the properties as you need them
+```
+
+### How can I use it?
+
+From the examples you could have a taste of the official SDK difficulties and how a higher-level library can help (I did not even mention [composing](https://developers.google.com/gmail/api/guides/sending#creating_messages) and [replying](https://stackoverflow.com/questions/32589476/how-to-send-a-reply-with-gmail-api/32591614#32591614) messages nor reading attachments).
```
Comment (alexantoniomoran) [inline content/posts/2020-07-01-introducing-gmail-wrapper/index.md:88]:
```suggestion
From the examples you could have a taste of the official SDK difficulties and how a higher-level library can help (We did not even mention [composing](https://developers.google.com/gmail/api/guides/sending#creating_messages) messages, [replying](https://stackoverflow.com/questions/32589476/how-to-send-a-reply-with-gmail-api/32591614#32591614) to messages, or reading attachments).
```

File: content/posts/2020-07-01-introducing-gmail-wrapper/index.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+If you felt interested you can have a more detailed documentation on GitHub.
```
Comment (alexantoniomoran) [inline content/posts/2020-07-01-introducing-gmail-wrapper/index.md:90]:
```suggestion
If you are interested in learning more about this package there is detailed documentation on the GitHub repository.
```

File: content/posts/2020-07-01-introducing-gmail-wrapper/index.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+To install, you already guess: `pip install gmail-wrapper`
```
Comment (alexantoniomoran) [inline content/posts/2020-07-01-introducing-gmail-wrapper/index.md:92]:
```suggestion
To install, you already guess it: `pip install gmail-wrapper`
```

File: content/posts/2020-07-01-introducing-gmail-wrapper/index.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+Of course, Gmail Wrapper has limitations too, like sending emails with attachments or authenticating with non-renewable tokens, for example, but these can be easily addressed by the community.
```
Comment (alexantoniomoran) [inline content/posts/2020-07-01-introducing-gmail-wrapper/index.md:94]:
```suggestion
Of course the initial build of our gmail-grapper has limitations. For example, sending emails with attachments or authenticating with non-renewable tokens are not possible. However, these are solvable issues that can easily be addressed by the community.
```

File: content/posts/2020-07-01-introducing-gmail-wrapper/index.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+So, don’t forget to contribute if you find any bugs or if you think something can be improved, we will be more than happy with your suggestion or pull-request.
```
Comment (alexantoniomoran) [inline content/posts/2020-07-01-introducing-gmail-wrapper/index.md:95]:
```suggestion
So please don’t forget to contribute if you find any bugs or if you think something can be improved! We will be more than happy to review your suggestions or pull-requests.

Thank you!
```

File: content/posts/2020-07-01-introducing-gmail-wrapper/index.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+### How can I use it?
+
+From the examples you could have a taste of the official SDK difficulties and how a higher-level library can help (We did not even mention [composing](https://developers.google.com/gmail/api/guides/sending#creating_messages) messages, [replying](https://stackoverflow.com/questions/32589476/how-to-send-a-reply-with-gmail-api/32591614#32591614) to messages, or reading attachments).
+
+If you are interested in learning more about this package there is detailed documentation on the GitHub repository.
+
+To install, you already guess it: `pip install gmail-wrapper`
+
+Of course the initial build of Gmail Wrapper has limitations. For example, sending emails with attachments or authenticating with non-renewable tokens are not possible.
```
Comment (alexantoniomoran) [inline content/posts/2020-07-01-introducing-gmail-wrapper/index.md:94]:
```suggestion
Of course the initial build of Gmail Wrapper has limitations. For example, sending emails with attachments or authenticating with non-renewable tokens is not possible.
```

Comment (luizguilhermefr) [reply content/posts/2020-07-01-introducing-gmail-wrapper/index.md]: Thank you!

**Comment by luizguilhermefr** [comment]
@barbosa when you want to merge 😉

**Pull request merged**